pkg/reconciler/route/resources/virtual_service_test.go

```diff
@@ -290,7 +307,72 @@ func ExampleMakeVirtualService() {
 		fmt.Printf("Regex %d: %s\n", i, h.Match[0].URI.Regex)
 	}
 
-	// Output: Regex 0: ^(/.*)?
+	// Output: Regex 0: ^/some-path-2(/.*)?
 	// Regex 1: ^/some-path-1(/.*)?
-	// Regex 2: ^/some-path-2(/.*)?
+	// Regex 2: ^(/.*)?
+}
+
+func ExampleMakeVirtualService_weightedRoutes() {
+	claims := []*v1alpha1.RouteClaim{
+		makeRouteClaim("some-host", "example.com/", "", "some-namespace"),
+		makeRouteClaim("some-host", "example.com/", "/path-a", "some-namespace"),
+		makeRouteClaim("some-host", "example.com/", "/path-b", "some-namespace"),
+	}
+
+	routes := []*v1alpha1.Route{
+		makeRoute("some-host", "example.com/", "", "some-app"),
+		makeRoute("some-host", "example.com/", "/path-a", "app-a-1"),
+		makeRoute("some-host", "example.com/", "/path-a", "app-a-2"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-1"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-2"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-3"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-4"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-5"),
+		makeRoute("some-host", "example.com/", "/path-b", "app-b-6"),
+	}
+
+	vs, err := resources.MakeVirtualService(claims, routes)
+	s, _ := json.MarshalIndent(vs, "", "\t")
```

Comment:
this JSON marshal doesn't seem to be adding to the output at all, I'd remove it for clarity

Reply:
oops forgot to remove after debugging, thanks for catching!